RecoTracker/DeDx/plugins/SiPixelTrackProbQXYProducer.cc

```diff
@@ -0,0 +1,189 @@
+      probXYonTrackWMulti *= probXY;
+
+    }  // end looping on the rechits
+
+    // Combine the probabilities into a track level quantity
+    float logprobQonTrackWMulti = probQonTrackWMulti > 0 ? log(probQonTrackWMulti) : 0;
+    float logprobXYonTrackWMulti = probXYonTrackWMulti > 0 ? log(probXYonTrackWMulti) : 0;
+    float factQ = -logprobQonTrackWMulti;
+    float factXY = -logprobXYonTrackWMulti;
+    float probQonTrackTerm = 1.f + factQ;
+    float probXYonTrackTerm = 1.f + factXY;
+
+    for (int iTkRh = 2; iTkRh < numRecHits; ++iTkRh) {
+      factQ *= -logprobQonTrackWMulti / float(iTkRh);
+      factXY *= -logprobXYonTrackWMulti / float(iTkRh);
+      probQonTrackTerm += factQ;
+      probXYonTrackTerm += factXY;
+    }
+
+    probQonTrack = probQonTrackWMulti * probQonTrackTerm;
+    probXYonTrack = probXYonTrackWMulti * probXYonTrackTerm;
+
+    // Repeat the above excluding Layer 1
+    float logprobQonTrackWMultiNoLayer1 = probQonTrackWMultiNoLayer1 > 0 ? log(probQonTrackWMultiNoLayer1) : 0;
+    float logprobXYonTrackWMultiNoLayer1 = probXYonTrackWMultiNoLayer1 > 0 ? log(probXYonTrackWMultiNoLayer1) : 0;
+
+    float factQNoLayer1 = -logprobQonTrackWMultiNoLayer1;
+    float factXYNoLayer1 = -logprobXYonTrackWMultiNoLayer1;
+    float probQonTrackTermNoLayer1 = 1.f + factQNoLayer1;
+    float probXYonTrackTermNoLayer1 = 1.f + factXYNoLayer1;
+
+    for (int iTkRh = 2; iTkRh < numRecHits; ++iTkRh) {
+      factQNoLayer1 *= -logprobQonTrackWMultiNoLayer1 / float(iTkRh);
+      factXYNoLayer1 *= -logprobXYonTrackWMultiNoLayer1 / float(iTkRh);
+      probQonTrackTermNoLayer1 += factQNoLayer1;
+      probXYonTrackTermNoLayer1 += factXYNoLayer1;
+    }
+
+    probQonTrackNoLayer1 = probQonTrackWMultiNoLayer1 * probQonTrackTermNoLayer1;
+    probXYonTrackNoLayer1 = probXYonTrackWMultiNoLayer1 * probXYonTrackTermNoLayer1;
+
+    indices.push_back(resultSiPixelTrackProbQXYColl->size());
+    resultSiPixelTrackProbQXYColl->emplace_back(
+        probQonTrack, probXYonTrack, probQonTrackNoLayer1, probXYonTrackNoLayer1);
+  }  // end loop on track collection
+
+  edm::OrphanHandle<reco::SiPixelTrackProbQXYCollection> siPixelTrackProbQXYCollHandle =
+      iEvent.put(std::move(resultSiPixelTrackProbQXYColl));
+
+  //create map passing the handle to the matched collection
+  auto trackProbQXYMatch = std::make_unique<reco::SiPixelTrackProbQXYAssociation>(siPixelTrackProbQXYCollHandle);
+  reco::SiPixelTrackProbQXYAssociation::Filler filler(*trackProbQXYMatch);
+  filler.insert(trackCollectionHandle, indices.begin(), indices.end());
+  filler.fill();
+  iEvent.put(std::move(trackProbQXYMatch));
+}
+
+void SiPixelTrackProbQXYProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.setComment("Producer that creates SiPixel Charge and shape probabilities combined for tracks");
+  desc.addUntracked<edm::InputTag>("tracks", edm::InputTag("isolatedTracks"))
```

Comment:
```suggestion
  desc.add<edm::InputTag>("tracks", edm::InputTag("isolatedTracks"))
```
this is set as untracked in the `fillDescriptions` but above is acquired as tracked.
https://github.com/cms-sw/cmssw/pull/36247/files#diff-819bbcd2c524b5b553e1970fc53af1f0392c52aecee75606cb5475eebdade73dR60.

I think it should be tracked.